core/testutils/src/main/java/com/mongodb/stitch/core/testutils/sync/SyncIntTestProxy.kt

```diff
@@ -1856,6 +1864,12 @@ class SyncIntTestProxy(private val syncTestRunner: SyncIntTestRunner) {
 
     private fun waitForAllStreamsOpen() {
         while (!syncTestRunner.dataSynchronizer.areAllStreamsOpen()) {
+            if (syncTestRunner
+                    .dataSynchronizer
+                    .getSynchronizedDocuments(syncTestRunner.namespace).size == 0) {
+                println("there are no documents to watch, so the stream will never open")
+                break
+            }
```

Comment:
Good catch here, surprised we never had tests that got tripped up on this

Reply:
yeah, we actually had tests that would hang once in a while, but this should resolve that